### PR TITLE
One build

### PR DESCRIPTION
**Pull request by bitdivine**
# Motivation
- Network errors are killing CI.
- The heaviest network usage is in building the canisters, either due to downloading dependencies or from getting the docker cache.
- So let's build once and reuse.

# Changes
* Move the `docker-build` and `sns_aggregator` workflows into `build.yaml`, where it can reuse the docker build.
* Add a workflow dispatch to `build.yaml`; this was available on docker-build.

Note:  I have studiously avoided making any "improvements" that are not strictly required for this PR.  Yes, to be sure, any reviewer is likely to find bits that could be improved.  But that is not the point of this PR.

# TODO
- After this has been merged, remove the empty workflows as merge requirements and make another PR deleting the workflows altogether.

# Tests
See how the docker build is now reused:

Screenshot from 2023-06-22 17-43-22

### CODE REVIEW

File: .github/workflows/aggregator.yaml
Comment (bitdivine) [inline .github/workflows/aggregator.yaml:1]: This code has been moved into build.yaml, where it can reuse the docker build there.

File: .github/workflows/build.yml
Comment (bitdivine) [inline .github/workflows/build.yml:1]: This workflow already built the assets once, then used the assets in several jobs:
Screenshot from 2023-06-22 17-49-02
This PR adds two more jobs, with the code taken from the docker-build and aggregator workflows:
Screenshot from 2023-06-22 17-43-22

File: .github/workflows/docker-build.yaml
Comment (bitdivine) [inline .github/workflows/docker-build.yaml:1]: This code has been moved into build.yaml

File: .github/workflows/checks.yml [hunk 1]
```diff
@@ -262,8 +262,14 @@ jobs:
         run: DFX_VERSION="$(jq -r .dfx dfx.json)" sh -ci "$(curl -fsSL https://sdk.dfinity.org/install.sh)"
       - name: Test getting proposal args
         run: scripts/dfx-nns-proposal-args.test
+  docker-build-cli-flags:
```
Comment (lmuntaner) [inline .github/workflows/checks.yml:265]: What does this do?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -41,6 +47,12 @@ jobs:
           name: sns_aggregator_dev
           path: out/sns_aggregator_dev.wasm.gz
           retention-days: 3
+      - name: 'Upload whole out directory'
```
Comment (lmuntaner) [inline .github/workflows/build.yml:50]: Why is this part needed?

**Review by lmuntaner — APPROVED** [review]
A couple of questions, already approving.

File: .github/workflows/build.yml (hunk 2 above)
Comment (bitdivine) [reply .github/workflows/build.yml:50]: The docker-build workflow publishes pretty much all the docker outputs.  Now, in build.yaml, the docker build is run but previously only the wasms were uploaded for reuse in other jobs.  So we need to upload more.

Comment (bitdivine) [reply .github/workflows/build.yml:50]: Thank you for the approval.  Closing to merge but happy to discuss further if wanted.

Comment (lmuntaner) [reply .github/workflows/build.yml:50]: Ok, thanks for the clarification!
